lib/sqlitex.ex

```diff
@@ -24,29 +24,46 @@ defmodule Sqlitex do
   {:ok, [%{a: 1, b: 2, c: 3}]}
 
   ```
+
+  ## Configuration
+
+  Sqlitex uses the Erlang library [esqlite](https://github.com/mmzeeman/esqlite)
+  which accepts a timeout parameter for almost all interactions with the database.
+  The default value for this timeout is 5000 ms. Many functions in Sqlitex accept
+  a timeout parameter that is passed on to the esqlite calls and that also defaults
+  to 5000 ms. If required, this default value can be overridden globally with the
+  following in your `config.exs`:
+
+  ```
+  config :sqlitex,
+    esqlite3_timeout: 10_000 # or other positive integer number of ms
+  ```
   """
 
-  @spec close(connection) :: :ok
-  def close(db) do
-    :esqlite3.close(db)
+  @esqlite3_timeout Application.get_env(:sqlitex, :esqlite3_timeout, 5_000)
+
+  @spec close(connection, integer()) :: :ok
+  def close(db, timeout \\ @esqlite3_timeout) do
+    :esqlite3.close(db, timeout)
   end
 
-  @spec open(charlist | String.t) :: {:ok, connection} | {:error, {atom, charlist}}
-  def open(path) when is_binary(path), do: open(string_to_charlist(path))
-  def open(path) do
-    :esqlite3.open(path)
+  @spec open(charlist | String.t, integer()) :: {:ok, connection} | {:error, {atom, charlist}}
+  def open(path, timeout \\ @esqlite3_timeout)
```

Comment:
I'm not sure, but would it be better to expose these parameters as keyword list?  That way if we need to add other options we can just add new keywords rather than needing to add a whole new parameter...

Reply:
I agree, I think it would be preferable. Will update accordingly.